### PR TITLE
dummy fall-back for non-linux platforms

### PR DESCRIPTION
**Pull request by havardpe**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@toregge please review

### CODE REVIEW

**Comment by baldersheim** [comment]
Test failure is unrelated

File: vespalib/src/vespa/vespalib/util/cpu_usage.h [hunk 1]
```diff
@@ -7,6 +7,19 @@ namespace vespalib {
...
+private:
+    steady_time _start;
+    double _load;
+public:
+    DummyThreadSampler();
+    void expected_load(double load);
+    duration sample() const;
+};
+
+#ifdef __linux__
```
Comment (baldersheim) [inline vespalib/src/vespa/vespalib/util/cpu_usage.h:21]: Is it possible to do this in the cpp file only to avoid ifdefs in the header files.

Comment (havardpe) [reply vespalib/src/vespa/vespalib/util/cpu_usage.h:21]: yes, but it will have a cost

Comment (havardpe) [reply vespalib/src/vespa/vespalib/util/cpu_usage.h:21]: what about an is_linux.h that defines a constexpr bool that we can use?

**Review by toregge — APPROVED** [review]

**Comment by havardpe** [comment]
@baldersheim @toregge PTAL

**Review by toregge — APPROVED** [review]

**Review by baldersheim — APPROVED** [review]
Perfect

**Pull request merged**